### PR TITLE
pingfederate_pingone_connection resource

### PR DESCRIPTION
**Pull request by wesleymccollam**

### CODE REVIEW

File: internal/acctest/config/pingoneconnection/ping_one_connection_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			}
+		},
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"pingfederate": providerserver.NewProtocol6WithError(provider.NewTestProvider()),
+		},
+		CheckDestroy: testAccCheckPingOneConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPingOneConnection(resourceName, initialResourceModel),
+				Check:  testAccCheckExpectedPingOneConnectionAttributes(initialResourceModel),
```
Comment (henryrecker-pingidentity) [inline internal/acctest/config/pingoneconnection/ping_one_connection_test.go:55]: Based on Patrick's recommendations, we should add some checks for computed attributes here

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+			},
+			"credential": schema.StringAttribute{
+				Description: "The credential for the PingOne connection.",
+				Optional:    true,
+				Sensitive:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"encrypted_credential": schema.StringAttribute{
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/pingoneconnection/ping_one_connection_resource.go:83]: We usually don't include these, right?

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+				Description: "The encrypted credential for the PingOne connection.",
+				Sensitive:   true,
+				Optional:    false,
+				Computed:    true,
+			},
+			"credential_id": schema.StringAttribute{
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/pingoneconnection/ping_one_connection_resource.go:89]: Do we want all of these in the resource? I know some other times we've left some of the read only fields for the data source. Not sure what is best.

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+	// Read the response
+	var state pingOneConnectionResourceModel
+	readPingOneConnectionResponse(ctx, updatePingOneConnectionResponse, &plan, &state)
+
+	// Update computed values
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// This config object is edit-only, so Terraform can't delete it.
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/pingoneconnection/ping_one_connection_resource.go:293]: Incorrect comment

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go (hunk 3 above)
Comment (wesleymccollam) [reply internal/resource/config/pingoneconnection/ping_one_connection_resource.go:89]: we left these in to match the API

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go (hunk 2 above)
Comment (wesleymccollam) [reply internal/resource/config/pingoneconnection/ping_one_connection_resource.go:83]: will remove this

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go (hunk 3 above)
Comment (henryrecker-pingidentity) [reply internal/resource/config/pingoneconnection/ping_one_connection_resource.go:89]: Ok, I had thought we had removed these in some other resources (key pairs) but it looks like we put them back in.

File: internal/acctest/config/pingoneconnection/ping_one_connection_test.go (hunk 1 above)
Comment (wesleymccollam) [reply internal/acctest/config/pingoneconnection/ping_one_connection_test.go:58]: added

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go (hunk 2 above)
Comment (wesleymccollam) [reply internal/resource/config/pingoneconnection/ping_one_connection_resource.go]: this is now removed

File: internal/resource/config/pingoneconnection/ping_one_connection_resource.go (hunk 4 above)
Comment (wesleymccollam) [reply internal/resource/config/pingoneconnection/ping_one_connection_resource.go]: now removed

File: internal/acctest/config/pingoneconnection/ping_one_connection_test.go [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+			},
+			{
+				// Test updating some fields
+				Config: testAccPingOneConnection(resourceName, updatedResourceModel),
+				Check:  resource.ComposeTestCheckFunc(testAccCheckExpectedPingOneConnectionAttributes(updatedResourceModel)),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue(fmt.Sprintf("pingfederate_ping_one_connection.%s", resourceName), tfjsonpath.New("creation_date"), knownvalue.NotNull()),
+					statecheck.ExpectKnownValue(fmt.Sprintf("pingfederate_ping_one_connection.%s", resourceName), tfjsonpath.New("credential_id"), knownvalue.NotNull()),
+					statecheck.ExpectKnownValue(fmt.Sprintf("pingfederate_ping_one_connection.%s", resourceName), tfjsonpath.New("environment_id"), knownvalue.StringExact("f5901536-2b60-4d4a-a987-3d56aadad46d")),
```
Comment (henryrecker-pingidentity) [inline internal/acctest/config/pingoneconnection/ping_one_connection_test.go:67]: Do we want this env id in the repo?

Comment (wesleymccollam) [reply internal/acctest/config/pingoneconnection/ping_one_connection_test.go]: edited to use the env var instead

**Review by henryrecker-pingidentity — APPROVED** [review]

**Pull request merged**